packages/jest-circus/src/index.js

```diff
@@ -38,12 +38,23 @@ const _dispatchDescribe = (blockFn, blockName, mode?: BlockMode) => {
   dispatch({blockName, mode, name: 'finish_describe_definition'});
 };
 
-const _addHook = (fn: HookFn, hookType: HookType, timeout: ?number) =>
-  dispatch({asyncError: new Error(), fn, hookType, name: 'add_hook', timeout});
-const beforeEach: THook = (fn, timeout) => _addHook(fn, 'beforeEach', timeout);
-const beforeAll: THook = (fn, timeout) => _addHook(fn, 'beforeAll', timeout);
-const afterEach: THook = (fn, timeout) => _addHook(fn, 'afterEach', timeout);
-const afterAll: THook = (fn, timeout) => _addHook(fn, 'afterAll', timeout);
+const _addHook = (fn: HookFn, hookType: HookType, hookFn, timeout: ?number) => {
+  const asyncError = new Error();
+  if (Error.captureStackTrace) {
+    Error.captureStackTrace(asyncError, hookFn);
+  }
+  dispatch({asyncError, fn, hookType, name: 'add_hook', timeout});
```

Comment:
it was returned previously? not sure if it matters

Reply:
`dispatch()` returns void.